src/main/java/com/jorgepatrick/NumberConverter.java

```diff
@@ -16,22 +19,29 @@ public NumberConverter(RomanNumberValidator romanNumberValidator) {
 
     public String parseArabicToRoman(final int arabicNumberSum) {
         String romanNumber = "";
-
-        int[] arabicDigits = new int[2];
         String arabicNumberString = String.valueOf(arabicNumberSum);
-        for(int i = 0; i < arabicNumberString.length(); i++) {
-            arabicDigits[i] = Character.digit(arabicNumberString.charAt(i), 10);
-        }
-        if (arabicNumberString.length() == 2){
-            romanNumber += setRomanDigit(arabicDigits[0], TEN);
-            romanNumber += setRomanDigit(arabicDigits[1], UNIT);
-        } else {
-            romanNumber += setRomanDigit(arabicDigits[0], UNIT);
+
+        int[] arabicDigits = splitNumberInReverse(arabicNumberSum, arabicNumberString.length());
+
+        for(int position = arabicNumberString.length() - 1; position >= 0; position--) {
```

Comment:
I didnt like this reverse for, seems to make the code harder to read, but not sure how to make it better @chicofaccin